connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -839,7 +840,22 @@ private void stopTask(ConnectorTaskId taskId) {
             ClassLoader savedLoader = plugins.currentThreadLoader();
             try {
                 savedLoader = Plugins.compareAndSwapLoaders(task.loader());
-                task.stop();
+                CountDownLatch latch = new CountDownLatch(1);
+                new Thread() {
+                    @Override
+                    public void run() {
+                        task.stop();
+                        latch.countDown();
+                    }
+                }.start();
+                // Wait for thread to terminate, but not longer than timeout.
+                if (timeout <= 0) {
```

Comment:
nit: This if statement is not required. The countdown latch won't wait at all if the timeout is <= 0.